railties/lib/rails/application.rb

```diff
@@ -368,6 +367,19 @@ def config=(configuration) #:nodoc:
       @config = configuration
     end
 
+    def sandbox(&block)
+      config.sandbox.inject(block) {|stack, klass|
+        -> do
+          begin
+            klass.start_sandbox
+            stack.call
+          ensure
+            klass.finish_sandbox
+          end
+        end
+      }.call
+    end
+
```

Comment:
We could grab all the connections in the sandbox then pass the list to a `Sandbox` instance to do the work:
```ruby
def sandbox(&block)
  connections = config.sandbox_classes.map { |k| k.constantize.connection }.uniq
  ActiveRecord::Sandbox.new(connections).start(&block)
end
```

We'll want to configure with class names as well, to account for dev-mode class reloading.